### ISSUE TITLE
Add Flower Baseline: FedBN - Convergence Rate

### ISSUE THREAD

**Issue by charlesbvll** [adap/flower#1744]
### Paper

Li et al., 2021, FedBN: Federated Learning on Non-IID Features via Local Batch Normalization, Convergence Rate

### Link

https://arxiv.org/abs/2102.07623

### For first time contributors

- [x]  Read the [`first contribution` doc](https://flower.dev/docs/first-time-contributors.html)
- [x]  Complete the Flower tutorial
- [x]  Read the Flower Baselines docs to get an overview:
    - [x]  [https://flower.dev/docs/using-baselines.html](https://flower.dev/docs/using-baselines.html)
    - [x]  [https://flower.dev/docs/contributing-baselines.html](https://flower.dev/docs/contributing-baselines.html)

### Prepare - understand the scope

- [X] Read the paper linked above
- [X] Create the directory structure in Flower Baselines (just the `__init__.py` files and a `README.md`)
- [X] Before starting to write code, write down all of the specs of this experiment in a README (dataset, partitioning, model, number of clients, all hyperparameters, …)
- [X] Open a draft PR

### Implement - make it work

- [X] Implement some form of dataset loading and partitioning in a separate `dataset.py` (doesn’t have to match the paper exactly)
- [X] Implement the model in PyTorch
- [X] Write a test that shows that the model has the number of parameters mentioned in the paper
- [X] Implement the federated learning setup outlined in the paper, maybe starting with fewer clients
- [X] Plot accuracy and loss
- [X] Run it and check if the model starts to converge

### Align - make it converge

- [X] Implement the exact data partitioning outlined in the paper
- [X] Use the exact hyperparameters outlined in the paper
- [X] Make it converge to roughly the same accuracy that the paper states
- [X] Commit the final hyperparameters and plots
- [X] Mark the PR as ready
